hack/lib/test.sh

```diff
@@ -126,12 +128,12 @@ kube::test::get_object_jsonpath_assert() {
 kube::test::describe_object_assert() {
   local resource=$1
   local object=$2
-  local matches=${@:3}
+  local matches=( "${@:3}" )
 
-  result=$(eval kubectl describe "${kube_flags[@]}" ${resource} ${object})
+  result=$(eval kubectl describe "${kube_flags[@]}" "${resource}" "${object}")
 
-  for match in ${matches}; do
-    if [[ ! $(echo "${result}" | grep ${match}) ]]; then
+  for match in "${matches[@]}"; do
+    if echo "${result}" | grep -q "${match}"; then
```

Comment:
Fix for this:

```
In ./hack/lib/test.sh line 134:
    if [[ ! $(echo "${result}" | grep ${match}) ]]; then
            ^-- SC2143: Use grep -q instead of comparing output with [ -n .. ].
                                      ^------^ SC2086: Double quote to prevent globbing and word splitting.
```

Reply:
We've had issues piping to `grep -q` and use a heredoc `<<<` instead.

https://github.com/kubernetes/kubernetes/pull/74639#discussion_r279069215

Reply:
Noted, will update